Sources/Swift/Integrations/SessionReplay/SentryTouchTracker.swift

```diff
@@ -0,0 +1,149 @@
+import Foundation
+#if (os(iOS) || os(tvOS)) && !SENTRY_NO_UIKIT
+import UIKit
+
+@objcMembers
+class SentryTouchTracker: NSObject {
+    
+    private struct TouchEvent {
+        let x: CGFloat
+        let y: CGFloat
+        let timestamp: TimeInterval
+        let phase: TouchEventPhase
+        
+        var point: CGPoint {
+            CGPoint(x: x, y: y)
+        }
+    }
+    
+    private class TouchInfo {
+        let id: Int
+        
+        var start: TouchEvent?
+        var end: TouchEvent?
+        var movements = [TouchEvent]()
+        
+        init(id: Int) {
+            self.id = id
+        }
+    }
+    
+    private var trackedTouches = [UITouch: TouchInfo]()
+    private var touchId = 1
+    private let dateProvider: SentryCurrentDateProvider
+    private let scale: CGAffineTransform
+    
+    init(dateProvider: SentryCurrentDateProvider, scale: Float) {
+        self.dateProvider = dateProvider
+        self.scale = CGAffineTransform(scaleX: CGFloat(scale), y: CGFloat(scale))
+    }
+    
+    func trackTouchFrom(event: UIEvent) {
+        guard let touches = event.allTouches else { return }
+        for touch in touches {
+            guard touch.phase == .began || touch.phase == .ended || touch.phase == .moved || touch.phase == .cancelled else { continue }
+            let info = trackedTouches[touch] ?? TouchInfo(id: touchId++)
```

Comment:
`m`: `UITouch` doesn't conform to the `Hashable` protocol only via `NSObject`, so if I'm not mistaken the hash will be the memory address / pointer. Especially with touch events, I think two touch events might get the same memory address. What made you choose to store the touch event in a dict?

Reply:
That's the only way to know if the touch is the same. According to documentation the touch object remains the same in between events. While we hold reference to it no other touch event will get the same pointer address.

Reply:
Please add this info as a comment.